src/creo2urdf/src/Creo2Urdf.cpp

```diff
@@ -445,7 +441,12 @@ void Creo2Urdf::OnCommand() {
 bool Creo2Urdf::exportModelToUrdf(iDynTree::Model mdl, iDynTree::ModelExporterOptions options) {
     iDynTree::ModelExporter mdl_exporter;
 
-    mdl_exporter.init(mdl);
+    // Convert modelToExport in a URDF-compatible model (using the default base link)
+    iDynTree::Model modelToExportURDFCompatible;
+
+    bool ok = iDynTree::moveLinkFramesToBeCompatibleWithURDFWithGivenBaseLink(mdl, modelToExportURDFCompatible);
```

Comment:
I am a bit afraid in doing this unconditionally, as it could hide some bugs in the assignment of the frames in the models in which all the link frames are assigned. Perhaps we could do this only if link frames are not assigned, or have an explicit option to enable this mode that is off on models that do not need it?

Reply:
> Perhaps we could do this only if link frames are not assigned

Personally I'm in favour of this as it would avoid complicating the yaml.